File: stone_lib/src/main/java/com/github/klee0kai/stone/wrappers/creators/CircleWrapper.java
package com.github.klee0kai.stone.wrappers.creators;

public interface CircleWrapper extends ProviderWrapper {

    /**
     * UnProvide wrapped object.
     *
     * @param wrapperCl  type of wrapper
     * @param objectType original provide object type
     * @param wrapper    wrapper to unwrap
     * @param <Wr>       type of wrapper
     * @param <T>        type of providing original object
     * @return wrapped object provider
     */
    <Wr, T> T unwrap(Class<Wr> wrapperCl, Class<T> objectType, Wr wrapper);

}

File: stone_multiplatform/src/commonMain/kotlin/com/github/klee0kai/stone/wrappers/creators/CircleWrapper.kt
package com.github.klee0kai.stone.wrappers.creators

import kotlin.reflect.KClass

interface CircleWrapper : ProviderWrapper {
    /**
     * UnProvide wrapped object.
     *
     * @param wrapperCl  type of wrapper
     * @param objectType original provide object type
     * @param wrapper    wrapper to unwrap
     * @param <Wr>       type of wrapper
     * @param <T>        type of providing original object
     * @return wrapped object provider
    </T></Wr> */
    fun <Wr : Any, T : Any> unwrap(wrapperCl: KClass<Wr>?, objectType: KClass<T>?, wrapper: Wr?): T?
}
